jobs/index.js

```diff
@@ -86,10 +95,17 @@ function jobsClient(options) {
    //
    var jobSubscriptions = [];
 
-   //
-   // Instantiate the device
-   //
-   var device = deviceModule.DeviceClient(options);
+   // Allow reuse of the device client by passing options.device
+   if (!isUndefined(options.device) && typeof options.device === "object")
+   {
+      var device = options.device;
+   } else {
+      //
+      // Instantiate the device
+      //
+      var device = deviceModule.DeviceClient(options);
+
+   }
```

Comment:
Can you move the declaration of device outside the if and else blocks? It looks like local variable otherwise. Alternatively you could use the "?" operator.
``
var device = (!isUndefined(options.device) && typeof options.device === "object")
  ? options.device
  : deviceModule.DeviceClient(options);
``